### PR TITLE
Abfs 3.3.2 dev redirection dfs

### PR DESCRIPTION
**Pull request by saxenapranav**
test for:
1. fs.azure.enable.blob.endpoint
2. fs.azure.mkdirs.fallback.to.dfs
3. fs.azure.ingress.fallback.to.dfs
4. fs.azure.redirect.delete
5. fs.azure.redirect.rename

### CODE REVIEW

File: hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemBlobConfig.java [hunk 1]
```diff
@@ -0,0 +1,688 @@
...
+import static org.apache.hadoop.fs.azurebfs.constants.ConfigurationKeys.FS_AZURE_REDIRECT_RENAME;
+import static org.apache.hadoop.fs.azurebfs.constants.FileSystemConfigurations.ONE_MB;
+import static org.apache.hadoop.fs.azurebfs.constants.FileSystemUriSchemes.ABFS_DNS_PREFIX;
+import static org.apache.hadoop.fs.azurebfs.constants.FileSystemUriSchemes.WASB_DNS_PREFIX;
+
+public class ITestAzureBlobFileSystemBlobConfig
+    extends AbstractAbfsIntegrationTest {
+
+  public ITestAzureBlobFileSystemBlobConfig() throws Exception {
+    super();
```
Comment (snvijaya) [inline hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemBlobConfig.java:54]: should we add a Setup method that can skip this class if account is not FNS ?

**Review by snvijaya — CHANGES_REQUESTED** [review]
One minor comment on skipping test class for HNS account. Rest looks good to me.

File: hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemBlobConfig.java [hunk 2]
```diff
@@ -0,0 +1,688 @@
...
+      throws Exception {
+    AzureBlobFileSystem fs = createFileSystemForEndpointConfigPair(
+        FS_AZURE_ENABLE_BLOB_ENDPOINT, false, true);
+    AbfsClient client = Mockito.spy(fs.getAbfsClient());
+    fs.getAbfsStore().setClient(client);
+    fs.create(new Path("/tmp"));
+    veriCreatePathExecution(client);
+  }
+
+  private void veriCreatePathExecution(final AbfsClient client)
```
Comment (anmolanmol1234) [inline hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemBlobConfig.java]: Name should be verify ?

Comment (saxenapranav) [reply hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemBlobConfig.java]: taken.

File: hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemBlobConfig.java (hunk 1 above)
Comment (saxenapranav) [reply hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemBlobConfig.java:54]: added.

**Comment by saxenapranav** [comment]
Thanks. Have taken the comments.

**Review by snvijaya — APPROVED** [review]
LGTM

**Pull request merged**